gunicorn/workers/sync.py

```diff
@@ -126,8 +126,37 @@ def handle(self, listener, client, addr):
         req = None
         try:
             if self.cfg.is_ssl:
-                client = ssl.wrap_socket(client, server_side=True,
-                    **self.cfg.ssl_options)
+                # TODO: Make this reusable by gthread worker. Move to BaseWorker?
+                context = ssl.SSLContext(self.cfg.ssl_options['ssl_version'])
```

Comment:
I wonder if we can build this context once  when we parse the config and store it on `self.cfg`.

Reply:
Yes, building and storing it on `self.cfg` is definitely better than this :) We can move this snippet to `util.create_ssl_context()`and call it there.

I'm also planning to separate `Setting`s and `validate_*` functions in `gunicorn/config.py`.